CommunityEntity.UI.BezierEasing.cs

```diff
@@ -0,0 +1,137 @@
+using UnityEngine;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Facepunch.Extend;
+using System.IO;
+
+#if CLIENT
+
+public partial class CommunityEntity
+{
+
+    public class BezierEasing {
+        // this basically is the same process that Browsers use for their CSS easing, allowing you to define any possible easing value with 4 bezier points
+        // had to port this from a javascript version
+
+        public static Dictionary<string, BezierEasing> cache = new Dictionary<string, BezierEasing>();
+        public static int NewItts = 4;
+        public static float NewMinSlope = 0.001f;
+        public static float SubDivPrec = 0.0000001f;
+        public static float SubDivItts = 10;
+
+        public float mX1 = 0f;
+        public float mY1 = 0f;
+        public float mX2 = 1f;
+        public float mY2 = 1f;
+        public float[] mSampleValues;
+
+        public static int kSplineTableSize = 11;
+        public static float kSampleStepSize = 1.0f / (kSplineTableSize - 1.0f);
+        bool _precomputed = false;
+
+
+
+        public static float Ease(float X1, float Y1, float X2, float Y2, float time){
+            string key = $"{X1}{Y1}{X2}{Y2}";
+            if(!cache.ContainsKey(key)){
+                cache[key] = new BezierEasing(X1, Y1, X2, Y2);
+            }
+            return cache[key].GetPosition(time);
+        }
+
+
+
+        float A (float aA1, float aA2) { return 1.0f - 3.0f * aA2 + 3.0f * aA1; }
```

Comment:
1 letter methods are unreadable, especially when there is no documentation provided... pretty much never do this 😅

Reply:
i agree, in this case the three functions are a math equation broken up. i added a comment at the top but kept the variable names as changing them to "Part1/2/3" wouldnt have added any clarity that the comment doesnt allready adds
![image](https://user-images.githubusercontent.com/33698270/204253282-73391f43-be3a-4e91-9c37-f3e777719c3d.png)

> since i ported this from a js port of gecko's implementation i couldnt find a definition/description for what this formula is, otherwhise i would have changed the function names to include it